qa/rpc-tests/test_framework/util.py

```diff
@@ -603,13 +603,32 @@ def get_end_epoch_height(scid, node, epochLen):
     end_epoch_height = sc_creating_height - 1 + ((epoch_number + 1) * epochLen)
     return end_epoch_height
 
-def get_epoch_data(scid, node, epochLen):
-    sc_creating_height = node.getscinfo(scid)['items'][0]['createdAtBlockHeight']
-    current_height = node.getblockcount()
-    epoch_number = (current_height - sc_creating_height + 1) // epochLen - 1
-    end_epoch_block_hash = node.getblockhash(sc_creating_height - 1 + ((epoch_number + 1) * epochLen))
-    epoch_cum_tree_hash = node.getblock(end_epoch_block_hash)['scCumTreeHash']
-    return epoch_number, epoch_cum_tree_hash
+def get_epoch_data(scid, node, epochLen, is_non_ceasing = False, reference_height = None):
+    sc_info = node.getscinfo(scid)['items'][0]
+    last_cert_data_hash = 'PHANTOM_PREV_CERT_HASH'
+    if sc_info['state'] == 'UNCONFIRMED':
+        return None
+
+    if is_non_ceasing:
+        # For non-ceasing sidechains
+        epoch_number = sc_info['unconfTopQualityCertificateEpoch'] + 1 if 'unconfTopQualityCertificateEpoch' in sc_info else sc_info['epoch']
+        current_reference_height = node.getblockcount() if reference_height is None else reference_height
+        epoch_cum_tree_hash = node.getblock(str(current_reference_height))['scCumTreeHash']
+        if 'unconfTopQualityCertificateDataHash' in sc_info and sc_info['unconfTopQualityCertificateDataHash'] != '':
+            last_cert_data_hash = sc_info['unconfTopQualityCertificateDataHash']
+        elif 'lastCertificateDataHash' in sc_info and sc_info['lastCertificateDataHash'] != '':
+            last_cert_data_hash = sc_info['lastCertificateDataHash']
```

Comment:
Afer #497, this might be simplified:
```suggestion
        if 'lastCertificateDataHash' in sc_info and sc_info['lastCertificateDataHash'] != '':
            last_cert_data_hash = sc_info['lastCertificateDataHash']
```